src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -986,22 +779,440 @@ public void AnalyzeCommand_LoadConfigurationFile(string configValue, bool defaul
+                        }
+                    }
+                },
+                Message = new Message
+                {
+                    Arguments = new List<string> { filePath, fileName },
+                    Text = string.Format(@"Found an issue in {0} (full path is {1}", filePath, fileName)
+                }
+            };
+        }
+
+        [Fact]
+        public void AnalyzeCommandBase_GetFileNameFromUriWorks()
+        {
+            var sb = new StringBuilder();
+
+            var testCases = new Tuple<string, string>[]
+            {
+                new Tuple<string, string>(null, null),
+                new Tuple<string, string>(@"file.txt", "file.txt"),
+                new Tuple<string, string>(@".\file.txt", "file.txt"),
+                new Tuple<string, string>(@"c:\directory\file.txt", "file.txt"),
+                new Tuple<string, string>(@"\\computer\computer\file.txt", "file.txt"),
+                new Tuple<string, string>(@"file://directory/file.txt", "file.txt"),
+                new Tuple<string, string>(@"/file.txt", "file.txt"),
+                new Tuple<string, string>(@"directory/file.txt", "file.txt"),
+            };
+
+            foreach (Tuple<string, string> testCase in testCases)
+            {
+                Uri uri = testCase.Item1 != null ? new Uri(testCase.Item1, UriKind.RelativeOrAbsolute) : null;
+                string expectedFileName = testCase.Item2;
+
+                string actualFileName = AnalyzeCommandBase<TestAnalysisContext, AnalyzeOptionsBase>.GetFileNameFromUri(uri);
+
+                if (!object.Equals(actualFileName, expectedFileName))
+                {
+                    sb.AppendLine(string.Format("Bad file name returned for uri '{0}'. Expected '{1}' but saw '{2}'.", uri, expectedFileName, actualFileName));
+                }
+            }
+            sb.Length.Should().Be(0, because: "all URI to file name conversions should succeed but the following cases failed." + Environment.NewLine + sb.ToString());
+        }
+
+        #region ResultsCachingTestsAndHelpers
+
+        [Fact]
+        public void AnalyzeCommandBase_CachesErrors()
+        {
+            // Produce two errors results
+            var testCase = new ResultsCachingTestCase()
+            {
+                Files = new List<string>(new string[] { "Error.dll", "Error.exe" })
+            };
+
+            RunResultsCachingTestCase(testCase);
+
+            testCase.Verbose = true;
+            RunResultsCachingTestCase(testCase);
+        }
+
+
+        [Fact]
+        public void AnalyzeCommandBase_CachesNotes()
+        {
+            // Produce three results in verbose runs only
+            var testCase = new ResultsCachingTestCase()
+            {
+                Files = new List<string>(new string[] { "Note.dll", "Note.exe", "Note.sys" })
+            };
+
+            // Notes are verbose only results
+            testCase.ExpectedResultsCount.Should().Be(0);
```

Comment:
>ExpectedResultsCount [](start = 21, length = 20)

Oh now we're getting all meta, testing the test cases :-) It's a good idea of course, it just struck me funny.